Create DistributionBasePlugin (#32131)
gradle/gradle: platforms/documentation/docs/src/docs/release/notes.md

Nested events are reflected in the HTML test reports, providing clear traceability.

#### Distribution base plugin introduced

> This release introduces a new `distribution-base` plugin.

Reviewer: ```suggestion
Gradle now includes a `distribution-base` plugin, which mirrors the functionality of the [Distribution Plugin](userguide/distribution_plugin.html) but does not add a default distribution.
```

> This plugin is identical to the existing [Distribution Plugin](userguide/distribution_plugin.html), except it does not add any distributions by default.

Reviewer: ```suggestion

```

> The existing `distribution` plugin is now a simple wrapper over the `distribution-base` plugin, except it additionally adds a `main` distribution by default.

Reviewer: ```suggestion
The existing `distribution` plugin now acts as a wrapper for the `distribution-base` plugin, adding a default `main` distribution.
```

> The `distribution-base` plugin is useful when you want to create plugins that do not have a `main` distribution, while retaining the other features of the Distribution Plugin.

Reviewer: ```suggestion
The `distribution-base` plugin is particularly useful for plugin developers who want the capabilities of the Distribution Plugin without a `main` distribution.
```
